stable_plugins/classical_ml/data_preparation/pandas_preprocessing/backend/pandas_preprocessing.py

```diff
@@ -0,0 +1,309 @@
+    :param keep: str
+    :param ignore_index: bool
+    :return: DataFrame
+    """
+    subset = process_subset(subset)
+
+    keep = False if keep == "none" else keep
+
+    df.drop_duplicates(subset=subset, keep=keep, ignore_index=ignore_index, inplace=True)
+
+    return df
+
+
+def sort_values(df: DataFrame, by: str, ascending: bool = False, **kwargs):
+    """
+    Executes the sort_values function of pandas' dataframes (https://pandas.pydata.org/pandas-docs/version/1.5.0/reference/api/pandas.DataFrame.sort_values.html)
+     and returns the new dataframe.
+    :param df: DataFrame
+    :param by: str
+    :param ascending: str
+    :return: DataFrame
+    """
+    df.sort_values(by=by, ascending=ascending, inplace=True)
+    return df
+
+
+def strip_characters(
+    df: DataFrame,
+    characters: List[str] = "",
+    subset: str = None,
+    position: str = "both",
+    **kwargs,
+) -> DataFrame:
+    """
+    First transforms subset from a string into a list of strings and then continues to strip each entry in each given
+    column from the specified characters. If subset is empty, then all columns will be considered.
+    - position='front': A left strip will be used, to strip each entry.
+    - position='end': A right strip will be used, to strip each entry.
+    - position='both': Both left and right side of an entry will be striped.
+    :param df: DataFrame
+    :param subset: str containing the columns separated by commas
+    :param position: str
+    :return: DataFrame
+    """
+    subset = process_subset(subset)
+    if subset is None:
+        subset = df.keys()
+
+    function = None
+    if position == "front":
+        function = lambda s, c: s.lstrip(c)
+    elif position == "end":
+        function = lambda s, c: s.rstrip(c)
+    elif position == "both":
+        function = lambda s, c: s.strip(c)
+
+    for k in subset:
+        if k in df:
+            if {type(el) for el in df[k]} == {str}:
+                df[k] = function(df[k].str, characters)
+
+    return df
+
+
+def split_column(
+    df: DataFrame,
+    column: str,
+    by: str,
+    new_columns: str = "",
+    remove_column: bool = False,
+    **kwargs,
+) -> DataFrame:
+    """
+    Splits a column into many columns, using the string `by` to split each entry.
+    :param df: DataFrame
+    :param column: str contains the name of the old column
+    :param by: str
+    :param new_columns: str contains the names of the new columns separated by commas
+    :param remove_column: bool
+    :return: DataFrame
+    """
+    if column not in df.keys():
+        raise ValueError(f"The dataframe has no column {column}.")
+
+    new_columns = new_columns.split(",")
+    for n_c in new_columns:
+        if n_c in df.keys():
+            raise ValueError(
+                f"Columns {new_columns} already exist in the given dataframe."
+            )
+
+    df[new_columns] = df[column].str.split(by, expand=True)
+    if remove_column:
```

Comment:
maybe check if old column name is also used in new_columns (do not remove a new column...)